### PR TITLE
Side effects after taskable long running operation

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
1. Transform index operation to async call.
2. Add cluster info into notification toast.
3. Increase the coverage of job handlers.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by xluo-aws — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

File: public/JobHandler/callbacks/force_merge.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    {
+      title: ((
+        <>
+          Force merging <FormatResourcesWithClusterInfo resources={extras.sourceIndex} clusterInfo={extras.clusterInfo} /> has failed.
+        </>
+      ) as unknown) as string,
+      text: ((
+        <>
+          The force merge operation has taken more than one hour to complete. To see the latest status, use `GET /.tasks/_doc/
+          {extras.taskId}`
```
Comment (Hailong-am) [inline public/JobHandler/callbacks/force_merge.tsx:160]: It is better to use `GET /_tasks/<task_id>`, query system index is not recommended.

File: public/JobHandler/callbacks/open.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            <span key={item}>
+              {index > 0 && ", "}
+              <DetailLink key={item} index={item} clusterInfo={extras.clusterInfo} />
+            </span>
+          ))}{" "}
+          has timed out
+        </>
+      ) as unknown) as string,
+      text: ((
+        <>Opening the index has taken more than one hour to complete. To see the latest status, use `GET /.tasks/_doc/{extras.taskId}`</>
```
Comment (Hailong-am) [inline public/JobHandler/callbacks/open.tsx:113]: same here, better to use `GET /_tasks/<task_id>`

File: public/JobHandler/callbacks/open.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}>;
+
+export const callbackForOpen: CallbackType = async (job: OpenJobMetaData, { core }) => {
+  const extras = job.extras;
+  const commonService = new CommonService(core.http);
+
+  const tasksResult = await commonService.apiCaller<OpenTaskResult>({
+    endpoint: "transport.request",
+    data: {
+      path: `/.tasks/_doc/${extras.taskId}`,
```
Comment (Hailong-am) [inline public/JobHandler/callbacks/open.tsx:25]: query system index is not recommended. use `GET /_tasks/<task_id>` instead

File: public/JobHandler/callbacks/force_merge.tsx (hunk 1 above)
Comment (SuZhou-Joe) [reply public/JobHandler/callbacks/force_merge.tsx:160]: Good point, will adjust this.

File: public/JobHandler/callbacks/open.tsx (hunk 3 above)
Comment (SuZhou-Joe) [reply public/JobHandler/callbacks/open.tsx:25]: There is some difference between /.tasks/_doc/${extras.taskId} and /_tasks/<task_id>, will make it a task in 2.9, let's keep it in 2.8

Comment (cwperks) [reply public/JobHandler/callbacks/open.tsx:25]: The difference is pretty minimal:

```
➜  config git:(main) ✗ curl --cert ./kirk.pem --key ./kirk-key.pem -XGET https://localhost:9200/.tasks/_doc/0nozY80cTViD5VPloPwRYA:6475 -k
{"_index":".tasks","_id":"0nozY80cTViD5VPloPwRYA:6475","_version":1,"_seq_no":0,"_primary_term":1,"found":true,"_source":{"completed":true,"task":{"node":"0nozY80cTViD5VPloPwRYA","id":6475,"type":"transport","action":"indices:admin/open","description":"open indices [sample-index]","start_time_in_millis":1730387678257,"running_time_in_nanos":203230625,"cancellable":false,"cancelled":false,"headers":{}},"response":{"acknowledged":true,"shards_acknowledged":true}}}%
➜  config git:(main) ✗ curl --cert ./kirk.pem --key ./kirk-key.pem -XGET https://localhost:9200/_tasks/0nozY80cTViD5VPloPwRYA:6475 -k
{"completed":true,"task":{"node":"0nozY80cTViD5VPloPwRYA","id":6475,"type":"transport","action":"indices:admin/open","description":"open indices [sample-index]","start_time_in_millis":1730387678257,"running_time_in_nanos":203230625,"cancellable":false,"cancelled":false,"headers":{}},"response":{"acknowledged":true,"shards_acknowledged":true}}
```

**Pull request merged**